### PR TITLE
[no-ticket][risk=no] Run new e2e tests only on PRs

### PR DESCRIPTION
**Pull request by dmohs**
No ticket. Test configuration change.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -814,7 +814,8 @@ workflows:
       - playground
```
Comment (calbach) [inline .circleci/config.yml:814]: Should this also be filtered?

**Review by evrii — APPROVED** [review]
I am not familiar with this syntax. Do you have any documentation for it?

File: .circleci/config.yml (hunk 1 above)
Comment (dmohs) [reply .circleci/config.yml:814]: I think you are right, but I'll make a separate PR.

**Comment by dmohs** [comment]
@ericsong I'm not either, just stole the pattern. Single test failure, so merging. Assuming flaky.

**Comment by calbach** [comment]
@evrii If you're talking about the `<<: ...`, search for "YAML anchors".

If you're asking about the supported features of this Circle config file, see https://circleci.com/docs/configuration-reference

**Pull request merged**